Fix CI and README typos

jessp01 opened the pull request:
- Fix `gocritic` findings:
    * Error: sloppyLen: len(goExecutable) <= 0 can be len(goExecutable) == 0 (gocritic)
    * Error: wrapperFunc: use bytes.ReplaceAll method in `bytes.Replace(content, []byte("\r\n"), []byte("\n"), -1)` (gocritic)
- Update ENV matrix in accordance to what's used in the main `cobra` repo
- Update .golangci.yml (taken from the main `cobra` repo).
- Update `actions/checkout`, `actions/setup-go@v5` and `golangci/golangci-lint-action` versions

jessp01 committed:
Fix typos

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/spf13/cobra-cli?pullRequest=116) <br/>All committers have signed the CLA.

marckhouzam reviewed (approved):
Thanks!

marckhouzam commented:
I haven’t looked at the cobra-cli in a very long time. CI seems to be broken

jessp01 committed:
Fix CI

jessp01 commented:
Hi @marckhouzam ,

I've aligned `.github/workflows/test.yml` with the matrix you use for the `cobra` repo. It should pass now.
If you'd like, I can also add a GoReleaser config, I think it can be useful for this repo.

jessp01 commented:
Now the tests themselves are failing. See https://github.com/jessp01/cobra-cli/actions/runs/13499877160/job/37715453667#step:6:75

jessp01 committed:
Update copyright

jessp01 committed:
Fix CI

jessp01 committed:
Update `actions/checkout`, `actions/setup-go@v5` and
`golangci/golangci-lint-action` versions

jessp01 committed:
Update .golangci.yml (taken from the main cobra repo).

jessp01 committed:
Update .golangci.yml (taken from the main cobra repo).

jessp01 committed:
Fix:

  Error: sloppyLen: len(goExecutable) <= 0 can be len(goExecutable) == 0 (gocritic)
  Error: wrapperFunc: use bytes.ReplaceAll method in `bytes.Replace(content, []byte("\r\n"), []byte("\n"), -1)` (gocritic)

jessp01 commented:
All CI issues are now fixed: https://github.com/jessp01/cobra-cli/actions/runs/13500614133

jessp01 commented:
Hi @marckhouzam ,

Did you get a chance to review the CI fixes I applied? Don't mean to rush you, of course, just thought I'd tag you to make sure you're aware.

jessp01 commented:
Hi @marckhouzam ,

Just following up on this.

Cheers,
